firefox-extension/background.js

```diff
@@ -0,0 +1,40 @@
+const newTabIds = []
+
+// On create, we don't know the URL of the tab so we return focus to the
+// original tab and add the tabId to the list of new tabs. Once we know the URL,
+// we can either focus it or redirect it to the main browser
+browser.tabs.onCreated.addListener((tab) => {
+  if (tab.url !== 'about:blank') return
+  browser.tabs.update(1, { active: true })
+  newTabIds.push(tab.id)
+})
+
+// What about login through google or something like that?
+browser.tabs.onUpdated.addListener((tabId, { url }) => {
```

Comment:
You might want to take a look at this instead:
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webNavigation/onBeforeNavigate